javalib/src/main/scala/java/util/regex/Matcher.scala

```diff
@@ -85,8 +85,8 @@ final class Matcher private[regex] (var _pattern: Pattern,
         var i = 0
         while (i < nMatches) {
           val m     = matches + i
-          val start = (m.data - in).toInt
-          val end   = start + m.length
+          val start = if (m.data == null) -1 else (m.data - in).toInt
+          val end   = if (m.data == null) -1 else start + m.length
```

Comment:
I'm not sure if there is one more case when re2 specifies open range as a group, like (2, inf) means the rest of the source string starts from 2th position.
There is also (inf, 5) means take(5). I have seen that kind of cases as well. But I just fixed deterministic here, I agree to fix more correctly.